examples/volume_plane_rendering.py

```diff
@@ -0,0 +1,226 @@
+    plane_layer.plane.thickness = widget.thickness_spinbox.value()
+
+
+def rotate_plane(axis: int, positive: bool, angle: float = 1.5):
+    current_normal = np.array(viewer.layers['plane'].plane.normal_vector).reshape(-1, 3,
+                                                                                  1)
+    normalised = current_normal / np.linalg.norm(current_normal)
+
+    axis_str = 'xyz'[axis]
+    if not positive:
+        angle = -angle
+
+    new_normal = R.from_euler(axis_str, [angle], degrees=True).as_matrix() @ normalised
+    update_plane_normal(new_normal.reshape(-1))
+
+
+def shift_plane_along_normal(positive: bool = True):
+    plane_parameters = viewer.layers['plane'].plane
+    position = np.array(plane_parameters.position, dtype=float)
+    normal = np.array(plane_parameters.normal_vector, dtype=float)
+    shift = normal / np.linalg.norm(normal)
+
+    if not positive:
+        shift = -shift
+
+    new_position = position + shift
+    update_plane_position(new_position)
+
+
+def create_plane_widget():
+    widget = PlaneWidget()
+
+    widget.position_0.valueChanged.connect(lambda: update_plane_position(widget.plane_position))
+    widget.position_1.valueChanged.connect(lambda: update_plane_position(widget.plane_position))
+    widget.position_2.valueChanged.connect(lambda: update_plane_position(widget.plane_position))
+
+    widget.normal_0_button.clicked.connect(lambda: update_plane_normal((1, 0, 0)))
+    widget.normal_1_button.clicked.connect(lambda: update_plane_normal((0, 1, 0)))
+    widget.normal_2_button.clicked.connect(lambda: update_plane_normal((0, 0, 1)))
+    widget.normal_oblique_button.clicked.connect(lambda: update_plane_normal((1, 1, 1)))
+
+    widget.normal_rotate_0_positive_button.clicked.connect(lambda: rotate_plane(0, positive=True))
+    widget.normal_rotate_0_negative_button.clicked.connect(lambda: rotate_plane(0, positive=False))
+    widget.normal_rotate_1_positive_button.clicked.connect(lambda: rotate_plane(1, positive=True))
+    widget.normal_rotate_1_negative_button.clicked.connect(lambda: rotate_plane(1, positive=False))
+    widget.normal_rotate_2_positive_button.clicked.connect(lambda: rotate_plane(2, positive=True))
+    widget.normal_rotate_2_negative_button.clicked.connect(lambda: rotate_plane(2, positive=False))
+
+    widget.random_normal_button.clicked.connect(
+        lambda: update_plane_normal(np.random.uniform(low=0, high=1, size=3)))
+    widget.random_position_button.clicked.connect(
+        lambda: update_plane_position(np.random.uniform(low=0, high=64, size=3)))
+    widget.shift_normal_positive_button.clicked.connect(
+        lambda: shift_plane_along_normal(positive=True))
+    widget.shift_normal_negative_button.clicked.connect(
+        lambda: shift_plane_along_normal(positive=False))
+
+    widget.thickness_spinbox.valueChanged.connect(
+        lambda: update_plane_thickness(widget))
+    return widget
+
+
+plane_widget = create_plane_widget()
```

Comment:
I'm getting a segfault when i run this

```
(base) czirwc1macos2701:napari nsofroniew$ python examples/volume_plane_rendering.py
WARNING: QWidget: Must construct a QApplication before a QWidget
Abort trap: 6
```

Do you get one too? Maybe we need to delay widget creation?

Reply:
whoops! that's what I get for always running in ipython, will fix

Reply:
this now works best for me with the `app = get_app()` - as is nothing launches for me, and i don't see any error messages (might be because i have some error handing stuff turned off). @tlambert03 can you check this out too, thanks! I'll review rest of code in meantime

Reply:
I've dropped the `get_app` and it all works great for me. Hope that's ok, but with it still there i wouldn't see anything

Reply:
dropping should be fine as long as the creation of the Viewer comes before the creation of the plane widget.  (@sofroniewn, it's conceivable that you might have been seeing [this problem](https://github.com/napari/napari/issues/2925))